### DISCUSSION TITLE
Scoopy only shows a half wave

### DISCUSSION THREAD

**Discussion by 123kkkww** [fhdm-dev/scoppy#96] [Q&A]
Hello,
I tried Scoppy for the first time today.
On my smartphone, Scoppy only shows a half wave for all waveforms. I connected a function generator to channel 1. For example, you want to display a 50Hz sine wave. As the screenshot shows, only a half wave is displayed.
What am I doing wrong ?

![Screenshot_Scoppy](https://user-images.githubusercontent.com/40492137/221626260-c0c8ed50-2810-459e-beff-b55ac8f5d7ea.jpg)

**Comment by fhdm-dev** [comment]
Hi,

The Pico can only measure voltages between 0 and 3.3V. If your input voltage is outside this range you'll need to add some additional circuitry. See the section [Measuring different voltage ranges](https://github.com/fhdm-dev/scoppy#measuring-different-voltage-ranges-oscilloscope-mode) on the Scoppy home page and the [Analog Front-End](https://oscilloscope.fhdm.xyz/wiki/Analog-Front-End) page in the Documentation.

The DroneBot workshop YouTube channel has a video showing how you can [measure negative voltages](https://www.youtube.com/watch?v=8ldxmyujHK8&t=4523s) with Scoppy.

I hope that helps :smile:

**Reply by 123kkkww** [reply]
Hi,
As you can see on the screenshot, my max voltage is 3.29 volts and is therefore within the permissible range. But even at 2.8 volts, only half waves are displayed.
